algebra/pi_instances.lean

```diff
@@ -252,14 +256,28 @@ by constructor; simp [inl, inr] {contextual := tt}
 
 instance [has_scalar α β] [has_scalar α γ] : has_scalar α (β × γ) := ⟨λa p, (a • p.1, a • p.2)⟩
 
-instance {r : ring α} [module α β] [module α γ] : module α (β × γ) :=
-{ smul_add  := assume a p₁ p₂, mk.inj_iff.mpr ⟨smul_add, smul_add⟩,
-  add_smul  := assume a p₁ p₂, mk.inj_iff.mpr ⟨add_smul, add_smul⟩,
-  mul_smul  := assume a₁ a₂ p, mk.inj_iff.mpr ⟨mul_smul, mul_smul⟩,
-  one_smul  := assume ⟨b, c⟩, mk.inj_iff.mpr ⟨one_smul, one_smul⟩,
+@[simp] theorem smul_fst [has_scalar α β] [has_scalar α γ]
+  (a : α) (x : β × γ) : (a • x).1 = a • x.1 := rfl
+@[simp] theorem smul_snd [has_scalar α β] [has_scalar α γ]
+  (a : α) (x : β × γ) : (a • x).2 = a • x.2 := rfl
+@[simp] theorem smul_mk [has_scalar α β] [has_scalar α γ]
```

Comment:
what about `smul_pair`?

Reply:
well it's in the prod namespace already, and this is called `prod.mk`. It might also name conflict with the `pair` operator on linear maps (which is like `prod.mk` but for morphisms)